sdk/servicebus/Microsoft.Azure.ServiceBus/src/Primitives/ManagedIdentityTokenProvider.cs

```diff
@@ -12,7 +12,17 @@ namespace Microsoft.Azure.ServiceBus.Primitives
     /// </summary>
     public class ManagedIdentityTokenProvider : TokenProvider
     {
-        static AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+        private readonly AzureServiceTokenProvider azureServiceTokenProvider;
+
+        /// <summary>Initializes new instance of <see cref="ManagedIdentityTokenProvider"/> class with default <see cref="AzureServiceTokenProvider"/> configuration.
+        public ManagedIdentityTokenProvider() : this(new AzureServiceTokenProvider()){}
```

Comment:
If the majority use case would be not injecting a custom provider, should we retain a static `AzureServiceTokenProvider` instance that can be lazily instantiated and shared for that case rather than altering the existing behavior and creating a new one for each `ManagedIdentityTokenProvider` instance?

Reply:
It used to be a static field. Lazy would be an improvement.